pkg/lib/authn/authenticator/service/service_readonly.go

```diff
@@ -0,0 +1,134 @@
+package service
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/api/model"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+)
+
+// RateLimiter depends on EventService
+// EventService depends on UserInfoService
+// So finally depends on authenticator.Service causing circular dependency
+// This service was created for read only methods and do not depends on RateLimiter to break this circular dependency
+type ReadOnlyService struct {
+	Store    *Store
+	Password PasswordAuthenticatorProvider
+	Passkey  PasskeyAuthenticatorProvider
+	TOTP     TOTPAuthenticatorProvider
+	OOBOTP   OOBOTPAuthenticatorProvider
+}
+
+func (s *ReadOnlyService) List(ctx context.Context, userID string, filters ...authenticator.Filter) ([]*authenticator.Info, error) {
```

Comment:
After moving the functions here, maybe we can remove the implementation in `pkg/lib/authn/authenticator/service/service.go`?

Reply:
Right, let me remove the duplicated method.